### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by paodb**

### CODE REVIEW

**Comment by mlopezFC** [comment]
I reviewed the addon and everything LGTM

File: src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/CustomDriverOptions.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import org.apache.commons.lang3.StringUtils;
+import elemental.json.Json;
+import elemental.json.JsonObject;
+
+/**
+ * Representation of the options to be defined to create a custom driver.
+ */
+public class CustomDriverOptions {
+
+  public String buttonText;
```
[the remark this answers is not in the document]
Comment (mlopezFC) [reply src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/CustomDriverOptions.java:32]: why not using getters / setters?

[the remark this answers is not in the document]
Comment (flang) [reply src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/CustomDriverOptions.java:32]: If this class is part of the API, instance variables should be private. Accessible using getters and setters only. Fluent setters could be used.

**Comment by flang** [comment]
LGTM

File: src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/LanguageKeys.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * Representation of the default language keys in Sharee library.
+ */
+public class LanguageKeys {
+
+  enum Direction {
+
+    LTR, RTL;
+  }
+
+  public String copiedSuccessfully = "Copied successfully";
```
Comment (javier-godoy) [inline src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/LanguageKeys.java:36]: The same observation applies https://github.com/FlowingCode/ShareEasy/pull/1#discussion_r1302133580

File: src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/Options.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+/**
+ * Representation of the default options that can be configure for a Share Easy component.
+ */
+public class Options {
+
+  /* By default current page's title */
+  public String shareText;
+
+  /* By default current page's url */
+  public String shareLink;
```
Comment (javier-godoy) [inline src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/Options.java:39]: The same observation applies https://github.com/FlowingCode/ShareEasy/pull/1#discussion_r1302133580

**Review by javier-godoy — CHANGES_REQUESTED** [review]

File: src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/CustomDriverOptions.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (javier-godoy) [reply src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/CustomDriverOptions.java]: Resolved.

File: src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/LanguageKeys.java (hunk 2 above)
Comment (paodb) [reply src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/LanguageKeys.java]: Done.

File: src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/Options.java (hunk 3 above)
Comment (paodb) [reply src/main/java/com/flowingcode/vaadin/addons/shareeasy/util/Options.java]: Done.

**Review by javier-godoy — APPROVED** [review]

**Review by mlopezFC — APPROVED** [review]
LGTM

**Review by flang — APPROVED** [review]
LGTM

**Pull request merged**